src/preact/component/iframe.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper, useValueRef} from '../../../src/preact/component';
+import {Loading} from '../../../src/core/loading-instructions';
+import {ReadyState} from '../../../src/ready-state';
+import {dict} from '../../../src/utils/object';
+import {forwardRef} from '../../../src/preact/compat';
+import {useAmpContext, useLoading} from '../../../src/preact/context';
+import {
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from '../../../src/preact';
+
+const NO_HEIGHT_STYLE = dict();
+
+/**
+ * @param {!IframeDef.Props} props
+ * @param {{current: (!IframeDef.Api|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+export function IframeEmbedWithRef(
+  {
+    allow,
+    allowFullscreen,
+    allowtransparency,
```

Comment:
I think to follow the fullscreen, maybe we should also pass this as `allowTransparency`?